vumi/blinkenlights/metrics.py

```diff
@@ -75,7 +75,8 @@ def stop_polling(self):
         Stop the metric polling and publishing task.
         """
         if self._task:
-            self._task.stop()
+            if self._task.running:
+                self._task.stop()
```

Comment:
One of the tests now tries to stop a task that was never started (because we're adding cleanup that wasn't there before) and that broke the manager teardown.